irohad/consensus/yac/storage/impl/yac_vote_storage.cpp

```diff
@@ -77,10 +82,37 @@ namespace iroha {
             break;
           case ProposalState::kSentNotProcessed:
             val = ProposalState::kSentProcessed;
-            break;
           case ProposalState::kSentProcessed:
-            break;
+            last_sent_processed_proposal_round_ =
+                last_sent_processed_proposal_round_
+                ? std::max(*last_sent_processed_proposal_round_, round)
+                : round;
+        }
+      }
+
+      boost::optional<Round> YacVoteStorage::getLastCompletedRound() const {
+        BOOST_ASSERT_MSG(
```

Comment:
Why do we need a terminating ASSERT here, which will disappear in the release version? I think, it's better to make a simple `if` and log error, if conditions are not met. It will also test an ability of the system to recover in case this error happens, which can be useful in release as well.

Reply:
Although this code is already removed, I am still sure that assert was the right decision, while re did not have a test covering this function. It was replacing this test while we ran the testing builds. And the condition it checks is guaranteed not to happen at runtime.